initfiles/bin/init_thorslave.in

```diff
@@ -44,24 +44,14 @@ slavename=thorslave_${hpcc_compname}
 stop_slaves()
 {
     local timer=15
-    local isAlive=0
-
-    log "Attempting to kill $slavename with SIGTERM"
-    killall -SIGTERM $slavename > /dev/null 2>&1
-    while [[ $isAlive -eq 0 && $timer -gt 0 ]];do
-        killall -0 $slavename > /dev/null 2>&1
-        isAlive=$?
-        [[ $isAlive -eq 0 ]] && sleep 0.5
-        ((timer--))
+    local _pidarray
+    local _pidname
+    
+    _pidarray=($( ls -l ${PID_DIR} | awk -v regpattern=${slavename}_[1-9][0-9]*\.pid '$0~regpattern {print $9;}' ))
+    for _pidname in ${_pidarray[@]}; do
+        kill_process ${PID_DIR}/$_pidname $slavename $timer
```

Comment:
Need to add PID_DIR since kill_process is expecting a full path

Reply:
pidwait_fn() has -
    if [[ $TIMEOUT -le 0 ]] && ps -p $WATCH_PID -o command= | grep "${PROCESS}" &>/dev/null ; then

Is this correct ?  Should it be if [[ $TIMEOUT -gt 0 ]] ?

Reply:
timeout should always be 1 or greater the way we're calling it. The kill_process function always calls pidwait_fn with a timeout of 1. At least that's the way I read it.

Reply:
Yea ok its fine as-is, TIMEOUT gets reduced above until its 0.   So checking its <= here is ok.

Reply:
Made the change suggested to ls -1.  Moved the awk back to grep since there's no more fields to process.  Tested locally.